exercises/circular-buffer/example.rs

```diff
@@ -4,36 +4,36 @@ pub enum Error {
     FullBuffer,
 }
 
-pub struct CircularBuffer {
-    buffer: Vec<char>,
+pub struct CircularBuffer<T: Default + Clone> {
+    buffer: Vec<T>,
     size: usize,
     start: usize,
     end: usize,
 }
 
-impl CircularBuffer {
+impl<T: Default + Clone> CircularBuffer<T> {
     // this circular buffer keeps an unallocated slot between the start and the end
     // when the buffer is full. 
-    pub fn new(size: usize) -> CircularBuffer {
+    pub fn new(size: usize) -> CircularBuffer<T> {
         CircularBuffer { 
-            buffer: vec!['0'; size + 1], 
+            buffer: vec![T::default(); size + 1], 
```

Comment:
Ah good idea about using `default`... when I was thinking about how I would tackle this I thought I might have to use `Option` which I suppose wouldn't be a terrible thing, but now you have taught me about the `Default` trait. Very good, thanks.